task_processing/plugins/kubernetes/task_config.py

```diff
@@ -238,6 +238,12 @@ def _valid_service_account_name(
     )
 
 
+def _float_or_none(str: Optional[str]) -> Optional[float]:
+    if str:
+        return float(str)
+    return None
```

Comment:
should this be something like:
```suggestion
def _float_or_none(val: Any) -> Optional[float]:
    return float(val) if val is not None else None
```

otherwise, the type hints aren't exactly correct (`_float_or_none(1)` is a valid call, but `1` is not a string.

thinking about this line also made me realize that I think we want an `invariant` set for the `*_request` fields as they should always be non-zero if set (I was going to say that the current factory would return `None` if passed in `0` or `0.0`, but then I realized that we don't really want 0-valued requests in the first place)

that said, I'm not quite sure what the ordering is here (i.e., is `factory()` called first or is `invariant()`?) so we'll probably need to do some testing here

Reply:
seems to run the invariant after factory i believe (?)